posix-mapper/src/main/java/org/opencadc/posix/mapper/web/PosixMapperAction.java

```diff
@@ -94,19 +94,23 @@
 
 public abstract class PosixMapperAction extends RestAction {
 
-    protected PosixClient posixClient;
     protected static final MultiValuedProperties POSIX_CONFIGURATION = PosixInitAction.getConfig();
     protected static final String TSV_CONTENT_TYPE = "text/tab-separated-values";
 
+    // As the Postgres class uses a Factory pattern, a static client is created to prevent multiple factory instances
+    // being created that was causing a memory leak.
+    // jenkinsd (2024.12.23)
+    //
+    protected static final PosixClient POSIX_CLIENT =
```

Comment:
Are you certain that PostgresPosixClient (and it's member Postgres) are 100% thread-safe? That's the risk of declaring a static (or storing a shared instance in JNDI) - it gets used by mutliple request threads and has to be thread-safe.

Reply:
True, and I'm not 100%.  I was able to use four clients hitting the service with three uid and gid entries without fail in a continuous loop.  The problem it solves is having a single Hibernate `SessionFactory` instance, which is thread safe:
https://docs.jboss.org/hibernate/core/3.5/api/org/hibernate/SessionFactory.html

I'm not sure if there is a better way to do this without Spring injecting it, or maybe having some JNDI lookup for it?

Reply:
As long as the trail of stateful objects are either immutable or themselves threadsafe, then it should be ok. 

Using JNDI instead of static doesn't make it more threadsafe, it just effects the scope of the instance:

static: whole JVM
JDNI: can be scoped to a single app or a single endpoint by how the JNDI key is specified

That doesn't matter here because this is service code, not library code and only deployed as a container with one app in the jvm.